### PR TITLE
Fixes #350: Fix for using JSON .->> operator within a restrict clause

### LINKED ISSUE #350: Using the Opaleye Json operator (.->>) in a restrict (within a runQuery) causes a fatal SQL error in runtime
Note : This issue occurs with PostgreSQL 9.4.4

Using the `.->>` operator to extract out a part of a PGJsonb column in a `restrict` gives the following error: 

```
SqlError {sqlState = "42883", sqlExecStatus = FatalError, sqlErrorMsg = "operator does not exist: jsonb ->> boolean", sqlErrorDetail = "", sqlErrorHint = "No operator matches the given name and argument type(s). You might need to add explicit type casts."}
```

The following is the code with which I encountered the error : 
``` 
runSearchQuery :: (HasDatabase m) => String -> m[FilteredJob]
runSearchQuery searchParam = runQuery $ proc() -> do
  result <- leftJoinQueryJobUserClient -< ()  
  restrict -< ((fromNullable (result^._5) (toNullable(result^._1).->>(pgStrictText "tag"))) .== (pgStrictText $ toS searchParam)) 
  returnA -< result
```

### PR DESCRIPTION
**Pull request by kahlil29**
Fix for issue #350 as suggested by @tomjaguarpaw .
Also includes a special (and specific) test for this fix.

### CODE REVIEW

**Comment by saurabhnanda** [comment]
👍  for `first-time contributor`

Btw, what's the PG version in the CI environment?

File: Test/Test.hs [hunk 1]
```diff
@@ -785,6 +785,14 @@ testJsonGetFieldText dataQuery = it "" $ testH q (`shouldBe` expected)
             Arr.returnA -< O.toNullable c1 O..->> O.pgStrictText "c"
         expected :: [Maybe T.Text]
         expected = [Just "21"]
+      
+-- Special Test for Github Issue #350
```
Comment (saurabhnanda) [inline Test/Test.hs]: Put the complete URL, probably?

Comment (tomjaguarpaw) [reply Test/Test.hs]: Yes please

**Comment by tomjaguarpaw** [comment]
> 👍 for first-time contributor

👍

**Comment by tomjaguarpaw** [comment]
> Btw, what's the PG version in the CI environment?

9.3:  https://github.com/tomjaguarpaw/haskell-opaleye/blob/master/.travis.yml#L24

I guess we could test on other versions too.

**Comment by kahlil29** [comment]
Updated the comment in Test.hs with the complete URL :)

**Comment by tomjaguarpaw** [comment]
Looks grand, thanks!

**Comment by kahlil29** [comment]
Hey @tomjaguarpaw , any reason why you closed this and did not merge?

**Comment by tomjaguarpaw** [comment]
Oh I thought I merged.  Thanks for the info.

**Pull request merged**